pkg/controller/planexecution/planexecution_controller.go

```diff
@@ -542,6 +434,148 @@ func (r *ReconcilePlanExecution) Reconcile(request reconcile.Request) (reconcile
 	return reconcile.Result{}, nil
 }
 
+// fatalError is representing type of error that is non-recoverable (like bug in the template preventing rendering)
+// we should not retry these errors
+type fatalError struct {
+	err error
+}
+
+func (e fatalError) Error() string {
+	return fmt.Sprintf("Fatal error: %v", e.err)
+}
+
+// populatePlanExecutionFromActivePlanDefinition reads content of the Plan defined in operator version and populates PlanExecution with data from rendered templates
+func populatePlanExecutionFromActivePlanDefinition(activePlan kudov1alpha1.Plan, planExecution *kudov1alpha1.PlanExecution, instance *kudov1alpha1.Instance, operatorVersion *kudov1alpha1.OperatorVersion, recorder record.EventRecorder) error {
```

Comment:
```suggestion
func populatePlanExecution(plan kudov1alpha1.Plan, pe *kudov1alpha1.PlanExecution, instance *kudov1alpha1.Instance, ov *kudov1alpha1.OperatorVersion, recorder record.EventRecorder) error {
```
Do we plan to populate from another source?   can the source be one of the params?   this function name seems too long... also this line of code is 243 chars long...

Reply:
okay! makes sense